src/worker.js

```diff
@@ -25,30 +25,50 @@ editModule("fs", (fs) => {
 
 const componentCompiler = require("@vue/component-compiler");
 
-module.exports = async ({ filename, source, trackUsedFiles }) => {
-  const compiler = componentCompiler.createDefaultCompiler();
+module.exports = async ({
+  filename,
+  source,
+  trackUsedFiles,
+  extractCss,
+  production,
+}) => {
+  const compilerOptions = production
+    ? { template: { isProduction: true } }
+    : {};
+  const compiler = componentCompiler.createDefaultCompiler(compilerOptions);
   usedFiles = new Set();
   try {
     if (/^\s*$/.test(source)) {
       throw new Error("File is empty");
     }
     const result = compiler.compileToDescriptor(filename, source);
+    let styles = [];
+
     const resultErrors = combineErrors(result.template, ...result.styles);
     if (resultErrors.length > 0) {
-      return { result: { errors: resultErrors }, usedFiles };
+      return { errors: resultErrors, usedFiles };
+    }
+    if (extractCss) {
+      styles = result.styles.map(({ code, scoped, media, module }) => ({
+        code,
+        scoped,
+        media,
+        module,
+      }));
+      result.styles = result.styles.map((style) =>
+        Object.assign(style, { code: "" })
+      );
```

Comment:
What's the reason to not just do `result.styles = []` here?

Reply:
If we remove the style items, the rendered Vue components will lose their scopeId information. I found that by only clearing the `code` property (which contains the CSS code) scopeId is preserved.

I've optimized a bit the code and added a short comment about this in https://github.com/apeschar/esbuild-vue/pull/6/commits/b793f4a609a6681f2982396dddb8c873549131df